crates/gpui/src/platform/linux/x11/window.rs

```diff
@@ -202,13 +202,22 @@ impl X11WindowState {
             })
             .unwrap();
 
-        // Is this ID used anywhere?
-        let idle_notify_event_id = xcb_connection.generate_id();
+        // Selects the set of Present events to be delivered for the specified window and event context.
         xcb_connection
             .send_and_check_request(&xcb::present::SelectInput {
-                eid: idle_notify_event_id,
+                eid: xcb_connection.generate_id(),
```

Comment:
what is the role of this ID that is generated but never used anywhere (other than this call) ?

Reply:
The event id is not used anywhere. It can be used to send another SelectInput if an app wishes to modify the events it is subscribed to. Also if an app wishes to get events several times then it can be used to differentiate the events that are sent back.